index/compaction.go

```diff
@@ -0,0 +1,319 @@
+package index
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/parquet-go/parquet-go"
+
+	"github.com/polarsignals/frostdb/dynparquet"
+	"github.com/polarsignals/frostdb/parts"
+)
+
+const (
+	IndexFileExtension     = ".idx"
+	ParquetCompactionTXKey = "compaction_tx"
```

Comment:
Do we still need this compaction tx logic for index files? Since we now hard link files in snapshots can we just rely on the snapshot txn to truncate the WAL as we previously did?

Reply:
Yes for a couple reasons. First is we maintain compaction tx on parts in general and while this isn't strictly used since read_tx have no function it's outside of the purview of this PR to change that behavior. Second is we still replay the index when there are no snapshots available, and require the compaction tx to correctly replay the index.

If we got rid of index replay and just fell back on the wal we could probably remove them.

Reply:
OK! I don't think we should get rid of index replay but I agree this doesn't belong in this PR.